internal/queue/loops/router/router.go

```diff
@@ -0,0 +1,158 @@
+/*
+Copyright (c) 2025 Diagrid Inc.
+Licensed under the MIT License.
+*/
+
+package router
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"sync/atomic"
+
+	"github.com/dapr/kit/ptr"
+	"github.com/diagridio/go-etcd-cron/internal/api/queue"
+	"github.com/diagridio/go-etcd-cron/internal/queue/actioner"
+	"github.com/diagridio/go-etcd-cron/internal/queue/loops"
+	"github.com/diagridio/go-etcd-cron/internal/queue/loops/counters"
+	"github.com/go-logr/logr"
+)
+
+type Options struct {
+	Actioner actioner.Interface
+	Log      logr.Logger
+	Cancel   context.CancelFunc
+}
+
+// router is a loop to handle Job events. It routes the event to the correct
+// counter loop. It creates and runs counter loops at inform instantiate time,
+// and garbage collects them after close.
+// TODO: @joshvanl: add sync.Pool for counters to reduce allocations.
+type router struct {
+	log    logr.Logger
+	cancel context.CancelFunc
+	act    actioner.Interface
+
+	counters map[string]*counter
+	wg       sync.WaitGroup
+}
+
+// counter is a single instance of a counter loop. Tracks the idx of the
+// associated key mod revision to track if the job event is still relevant for
+// the current live counter loop.
+type counter struct {
+	idx  *atomic.Int64
+	loop loops.Interface[*queue.JobAction]
+}
+
+func New(opts Options) loops.Interface[*queue.JobEvent] {
+	return loops.New(loops.Options[*queue.JobEvent]{
+		BufferSize: ptr.Of(uint64(1024)),
```

Comment:
Should this be a const somewhere? I recall seeing a buffer size in another file as well with this same value.

Reply:
No, the counters loop uses 5  https://github.com/diagridio/go-etcd-cron/pull/79/files#diff-9633f20fe5c8b6230518728488adc6ee3bc7f0ece430d865e23f36fb49234c64R49

See my comment above ^ https://github.com/diagridio/go-etcd-cron/pull/79#discussion_r2059242359